src/Zeebe.Client.Bootstrap/Extensions/ZeebeClientBuilderExtensions.cs

```diff
@@ -27,16 +27,18 @@ private static IZeebeClientFinalBuildStep BuildFinalStep(this IZeebeClientTransp
             if(options.TransportEncryption == null)
                 return builder.UsePlainText();
 
-            if(!String.IsNullOrEmpty(options.TransportEncryption.RootCertificatePath))
-                return builder.UseTransportEncryption(options.TransportEncryption.RootCertificatePath);
+            IZeebeSecureClientBuilder clientBuilder = null;
+            if (!String.IsNullOrEmpty(options.TransportEncryption.RootCertificatePath))
+                clientBuilder = builder.UseTransportEncryption(options.TransportEncryption.RootCertificatePath);
+            else
+                clientBuilder = builder.UseTransportEncryption();
```

Comment:
We could simplify in the Zeebe C# client the `builder.UseTransportEncryption` so it accepts null and defaults to the default encryption, would simplify this part. Maybe someone can create an issue for it.

Reply:
https://github.com/camunda-community-hub/zeebe-client-csharp/issues/385